rc/lsp.kak

```diff
@@ -1613,3 +1614,85 @@ def lsp-snippets-select-next-placeholders %{
         printf "select %s\n" "$selections"
     }
 }
+
+# grep filetype copied from kakoune with changes
+
+hook -group lsp-goto-highlight global WinSetOption filetype=lsp-goto %{
+    add-highlighter window/lsp-goto group
+    add-highlighter window/lsp-goto/ regex "^((?:\w:)?[^:\n]+):(\d+):(\d+)?" 1:cyan 2:green 3:green
+    add-highlighter window/lsp-goto/ line %{%opt{lsp_goto_current_line}} default+b
+    hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/lsp-goto }
+}
+
+hook global WinSetOption filetype=lsp-goto %{
+    hook buffer -group lsp-goto-hooks NormalKey <ret> lsp-goto-jump
+    hook -once -always window WinSetOption filetype=.* %{ remove-hooks buffer lsp-goto-hooks }
+}
+
+declare-option -docstring "name of the client in which all source code jumps will be executed" \
+    str jumpclient
+
+define-command -hidden lsp-goto-jump %{
+    evaluate-commands %{ # use evaluate-commands to ensure jumps are collapsed
+        try %{
+            execute-keys '<a-x>s^((?:\w:)?[^:]+):(\d+):(\d+)?<ret>'
+            set-option buffer lsp_goto_current_line %val{cursor_line}
+            evaluate-commands -try-client %opt{jumpclient} -verbatim -- edit -existing "%opt{lsp_goto_working_directory}/%reg{1}" %reg{2} %reg{3}
```

Comment:
This will fail on absolute paths - for example try `:lsp-references` on `App::new` in `src/main.rs`.
We can detect absolute paths:

```
try %{
    execute-keys -draft s^/<ret>
} catch %{
    set-register 1 "%opt{lsp_goto_working_directory}/%reg{1}"
}
evaluate-commands -try-client %opt{jumpclient} -verbatim -- edit -existing %reg{1} %reg{2} %reg{3}
```